src/hyp3_isce2/burst.py

```diff
@@ -0,0 +1,238 @@
+        f.write(content)
+
+    return str(out_file)
+
+
+def spoof_safe(asf_session: requests.Session, burst: BurstMetadata, base_path: Path = Path('.')) -> Path:
+    """Creates this file structure:
+    SLC.SAFE/
+    ├── manifest.safe
+    ├── measurement/
+    │   └── burst.tif
+    └── annotation/
+        ├── annotation.xml
+        └── calbiration/
+            ├── calibration.xml
+            └── noise.xml
+    """
+    safe_path = base_path / f'{burst.safe_name}.SAFE'
+    annotation_path = safe_path / 'annotation'
+    calibration_path = safe_path / 'annotation' / 'calibration'
+    measurement_path = safe_path / 'measurement'
+    paths = [annotation_path, calibration_path, measurement_path]
+    for p in paths:
+        if not p.exists():
+            p.mkdir(parents=True)
+
+    et_args = {'encoding': 'UTF-8', 'xml_declaration': True}
+
+    ET.ElementTree(burst.annotation).write(annotation_path / burst.annotation_name, **et_args)
+    ET.ElementTree(burst.calibration).write(calibration_path / burst.calibration_name, **et_args)
+    ET.ElementTree(burst.noise).write(calibration_path / burst.noise_name, **et_args)
+    ET.ElementTree(burst.manifest).write(safe_path / 'manifest.safe', **et_args)
+
+    burst_params = BurstParams(burst.safe_name, burst.swath, burst.polarization, burst.burst_number)
+    download_burst(asf_session, burst_params, measurement_path / burst.measurement_name)
+
+    return safe_path
+
+
+# TODO currently only validated for descending orbits
+def get_region_of_interest(
+    poly1: geometry.Polygon, poly2: geometry.Polygon, is_ascending: bool = True
+) -> Tuple[float, float, float, float]:
+    bbox1 = geometry.box(*poly1.bounds)
+    bbox2 = geometry.box(*poly2.bounds)
+    intersection = bbox1.intersection(bbox2)
+    bounds = intersection.bounds
+
+    x, y = (0, 1) if is_ascending else (2, 1)
+    roi = geometry.Point(bounds[x], bounds[y]).buffer(0.005)
+    bounds = roi.bounds  # returns (minx, miny, maxx, maxy)
+    return bounds
+
+
+def get_asf_session() -> requests.Session:
+    # requests will automatically use the netrc file:
+    # https://requests.readthedocs.io/en/latest/user/authentication/#netrc-authentication
+    session = requests.Session()
+    payload = {
+        'response_type': 'code',
+        'client_id': 'BO_n7nTIlMljdvU6kRRB3g',
+        'redirect_uri': 'https://auth.asf.alaska.edu/login',
+    }
+    response = session.get('https://urs.earthdata.nasa.gov/oauth/authorize', params=payload)
+    response.raise_for_status()
+    return session
+
+
+def download_bursts(param_list: Iterator[BurstParams]) -> List[BurstMetadata]:
+    """Steps
+    For each burst:
+        1. Download metadata
+        2. Create BurstMetadata object
+        3. Create directory structure
+        4. Write metadata
+        5. Download and write geotiff
+    """
+    bursts = []
+    with get_asf_session() as asf_session:
+        for i, params in enumerate(param_list):
```

Comment:
Earlier a variable `ii` was used, would be nice to choose one or the other for the sake of consistency.